### DISCUSSION TITLE
how to integrate provider with openssl no-shared static build

### DISCUSSION THREAD

**Discussion by vkosuri** [openssl/openssl#23085] [Q&A]
I have to build openssl static with no-shared, and I similarly built oqsprovider.so, I am trying to list oqsprovider, but it's not listing any of providers. Am I missing anything here? Will provider work with no-shared static?

OPENSSL_MODULES=. ./openssl-static list -provider oqsprovider -verbose

**Comment by levitte** [comment]
It's the same with a default shared library build:

``` console
$ openssl list -provider legacy -verbose
```

However, with the added option `-providers`, `openssl list` is told to list them:

``` console
$ openssl list -provider legacy -providers -verbose
Providers:
  legacy
    name: OpenSSL Legacy Provider
    version: 3.1.4
    status: active
    build info: 3.1.4
    gettable provider parameters:
      name: pointer to a UTF8 encoded string (arbitrary size)
      version: pointer to a UTF8 encoded string (arbitrary size)
      buildinfo: pointer to a UTF8 encoded string (arbitrary size)
      status: integer (arbitrary size)
```

So, try this and see what that does for you:

``` console
$ OPENSSL_MODULES=. ./openssl-static list -provider oqsprovider -providers -verbose
```

**Reply by vkosuri** [reply]
Thanks @levitte, Maybe I rewrite my question about how to link statically my ``oqsprovider`` built with ``./configure no-shared no-modules``
